actionview/test/template/sanitize_helper_test.rb

```diff
@@ -1,62 +1,298 @@
+#  - An integration test suite to verify basic expectations for each supported sanitizer vendor
+#
+
 require "abstract_unit"
 
-# The exhaustive tests are in the rails-html-sanitizer gem.
-# This tests that the helpers hook up correctly to the sanitizer classes.
+#
+#  Unit test SanitizeHelper behavior. We use a mock vendor to ensure we're testing behavior that is
+#  independent of sanitizer vendors.
+#
 class SanitizeHelperTest < ActionView::TestCase
   tests ActionView::Helpers::SanitizeHelper
 
-  def test_strip_links
-    assert_equal "Don't touch me", strip_links("Don't touch me")
-    assert_equal "on my mind\nall day long", strip_links("<a href='almost'>on my mind</a>\n<A href='almost'>all day long</A>")
-    assert_equal "Magic", strip_links("<a href='http://www.rubyonrails.com/'>Mag<a href='http://www.ruby-lang.org/'>ic")
-    assert_equal "My mind\nall <b>day</b> long", strip_links("<a href='almost'>My mind</a>\n<A href='almost'>all <b>day</b> long</A>")
-    assert_equal "&lt;malformed &amp; link", strip_links('<<a href="https://example.org">malformed & link</a>')
+  def setup
+    super
+    @saved_vendor = ActionView::Helpers::SanitizeHelper.sanitizer_vendor
+    @mock_vendor = ActionView::Helpers::SanitizeHelper.sanitizer_vendor = new_mock_vendor
   end
 
-  def test_sanitize_form
-    assert_equal "", sanitize("<form action=\"/foo/bar\" method=\"post\"><input></form>")
+  def teardown
+    ActionView::Helpers::SanitizeHelper.sanitizer_vendor = @saved_vendor
+    super
   end
 
-  def test_should_sanitize_illegal_style_properties
-    raw      = %(display:block; position:absolute; left:0; top:0; width:100%; height:100%; z-index:1; background-color:black; background-image:url(http://www.ragingplatypus.com/i/cam-full.jpg); background-x:center; background-y:center; background-repeat:repeat;)
-    expected = %r(\Adisplay:\s?block;\s?width:\s?100%;\s?height:\s?100%;\s?background-color:\s?black;\s?background-x:\s?center;\s?background-y:\s?center;\z)
-    assert_match expected, sanitize_css(raw)
+  test "sanitizer_vendor module attribute and class method" do
+    assert_equal(@mock_vendor, ActionView::Helpers::SanitizeHelper.sanitizer_vendor)
+    assert_equal(@mock_vendor, self.class.sanitizer_vendor)
+
+    vendor2 = ActionView::Helpers::SanitizeHelper.sanitizer_vendor = new_mock_vendor
+
+    assert_equal(vendor2, ActionView::Helpers::SanitizeHelper.sanitizer_vendor)
+    assert_equal(vendor2, self.class.sanitizer_vendor)
   end
 
-  def test_strip_tags
-    assert_equal("Don't touch me", strip_tags("Don't touch me"))
-    assert_equal("This is a test.", strip_tags("<p>This <u>is<u> a <a href='test.html'><strong>test</strong></a>.</p>"))
-    assert_equal "This has a  here.", strip_tags("This has a <!-- comment --> here.")
-    assert_equal("Jekyll &amp; Hyde", strip_tags("Jekyll & Hyde"))
-    assert_equal "", strip_tags("<script>")
+  [:full_sanitizer, :link_sanitizer, :safe_list_sanitizer].each do |method|
```

Comment:
Can we avoid dynamic test definitions? It makes it hard to debug and only run one of the cases instead of all 3

Reply:
Done